x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -0,0 +1,368 @@
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject(Connector.SERVICE_TYPE_FIELD.getPreferredName());
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject(Connector.STATUS_FIELD.getPreferredName());
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject(Connector.SYNC_NOW_FIELD.getPreferredName());
+                    builder.field("type", "boolean");
+                    builder.endObject();
+                }
+                builder.endObject();
+            }
+            builder.endObject();
+            return builder;
+        } catch (IOException e) {
+            logger.fatal("Failed to build " + CONNECTOR_CONCRETE_INDEX_NAME + " index mappings", e);
+            throw new UncheckedIOException("Failed to build " + CONNECTOR_CONCRETE_INDEX_NAME + " index mappings", e);
+        }
+    }
+
+    private static Settings getConnectorSyncJobsIndexSettings() {
+        return Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-3")
+            .put(IndexMetadata.SETTING_INDEX_HIDDEN, true)
+            .put("index.refresh_interval", "1s")
+            .build();
+    }
+
+    private static XContentBuilder getConnectorSyncJobsIndexMappings() {
+        try {
+            final XContentBuilder builder = jsonBuilder();
+            builder.startObject();
+            {
+                builder.startObject("_meta");
+                builder.field("version", Version.CURRENT.toString());
+                builder.endObject();
+
+                builder.field("dynamic", "false");
+                builder.startObject("properties");
+                {
+                    builder.startObject(ConnectorSyncJob.CANCELATION_REQUESTED_AT_FIELD.getPreferredName());
+                    builder.field("type", "date");
+                    builder.endObject();
+
+                    builder.startObject(ConnectorSyncJob.CANCELED_AT_FIELD.getPreferredName());
+                    builder.field("type", "date");
+                    builder.endObject();
+
+                    builder.startObject(ConnectorSyncJob.COMPLETED_AT.getPreferredName());
+                    builder.field("type", "date");
+                    builder.endObject();
+
+                    builder.startObject(ConnectorSyncJob.CREATED_AT_FIELD.getPreferredName());
+                    builder.field("type", "date");
+                    builder.endObject();
+
+                    builder.startObject(ConnectorSyncJob.CONNECTOR_FIELD.getPreferredName());
+                    {
+                        builder.startObject("properties");
```

Comment:
This object seems to be missing:
- `configuration`
- `filtering`
- `pipeline`